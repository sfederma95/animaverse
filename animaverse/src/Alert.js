import React from 'react';
import warning from './warning.svg'
import './alert.css';

function Alert({messages=[]}){
    console.log(typeof messages[0].response.data.error.msg)
    const errArr = messages.map(err=>{
        if (typeof err.response.data.error.msg === 'string'){
            return (
            <p id='err-container'><span key={err.response.data.error.msg} className='errors'>
            <img alt='warning error' className='err-img' src={warning}/>
            <span className='err-text'>{err.response.data.error.msg.toUpperCase()}.</span>
            </span></p>)
        } else {
            return <p id='err-container'>{err.response.data.error.msg.map(e=>(<span key={e} className='errors'>
            <img alt='warning error' className='err-img' src={warning}/>
            <span className='err-text'>{e.toUpperCase()}.</span>
            </span>))}</p>
        }
    })
    return(
        <div>
            {errArr}
        </div>
    )
}

export default Alert;