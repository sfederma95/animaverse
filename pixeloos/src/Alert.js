import React, {useEffect} from 'react';
import warning from './warning.svg'
import './alert.css';
import { v4 as uuidv4 } from 'uuid';
import targetErrInputs from './users/helper'

function Alert({messages=[]}){
    useEffect(()=>{
        targetErrInputs(messages)
    },[messages])
    const errArr = messages.map(err=>{
        if (typeof err.response.data.error.msg === 'string'){
            return (
            <p key={uuidv4()} id='err-container'><span key={err.response.data.error.msg} className='errors'>
            <img alt='warning error' className='err-img' src={warning}/>
            <span className='err-text'>{err.response.data.error.msg.toUpperCase()}.</span>
            </span></p>)
        } else {
            return <p key={uuidv4()} id='err-container'>{err.response.data.error.msg.map(e=>(<span key={e} className='errors'>
            <img alt='warning error' className='err-img' src={warning}/>
            <span className='err-text'>{e.toUpperCase()}.</span>
            </span>))}</p>
        }
    })
    return(
        <div>
            {errArr}
        </div>
    )
}

export default Alert;