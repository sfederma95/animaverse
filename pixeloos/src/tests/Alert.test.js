import React from 'react';
import {render} from '@testing-library/react';
import Alert from '../Alert';

it ('renders without crashing',function(){
    render(<Alert />)
})

it ('matches snapshot with one message',function(){
    let messages = [{
        response:{
            data:{
                error:{
                    msg: 'Invalid password/username'
                }
            }
        }
    }]
    const {asFragment} = render(<Alert messages={messages} />)
    expect(asFragment()).toMatchSnapshot();
})

it ('matches snapshot with multiple messages',function(){
    let messages = [{
        response:{
            data:{
                error:{
                    msg: ['Invalid password/username','Email must be in email format']
                }
            }
        }
    }]
    const {asFragment} = render(<Alert messages={messages} />)
    expect(asFragment()).toMatchSnapshot();
})