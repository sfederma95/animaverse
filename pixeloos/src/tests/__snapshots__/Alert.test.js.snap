// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with multiple messages 1`] = `
<DocumentFragment>
  <div>
    <p
      id="err-container"
    >
      <span
        class="errors"
      >
        <img
          alt="warning error"
          class="err-img"
          src="warning.svg"
        />
        <span
          class="err-text"
        >
          INVALID PASSWORD/USERNAME.
        </span>
      </span>
      <span
        class="errors"
      >
        <img
          alt="warning error"
          class="err-img"
          src="warning.svg"
        />
        <span
          class="err-text"
        >
          EMAIL MUST BE IN EMAIL FORMAT.
        </span>
      </span>
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot with one message 1`] = `
<DocumentFragment>
  <div>
    <p
      id="err-container"
    >
      <span
        class="errors"
      >
        <img
          alt="warning error"
          class="err-img"
          src="warning.svg"
        />
        <span
          class="err-text"
        >
          INVALID PASSWORD/USERNAME.
        </span>
      </span>
    </p>
  </div>
</DocumentFragment>
`;
